<a href="https://colab.research.google.com/github/gilesitorr/Estadistica_Avanzada/blob/main/Tarea_3/EA_tarea_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# __Estadística Avanzada__
# Tarea 3
## _Gil Estéfano Rodríguez Rivera_

__Nota:__ En este Notebook se usó el kernel de R, aunque la extensión sea ipynb.

__Nota:__ También puede encontrar el desarrollo en _GitHub_ en https://github.com/gilesitorr/Estadistica_Avanzada/tree/main/Tarea_3

## Introduction to Probability and Statistics for Engineers and Scientists
### Sheldon M. Ross

### 42


__Ten pregnant women given an injection of pitocin to induce labor. Their systolic blood pressures immediately before and after the injection were:__

_Before_: 134 122 132 130 128
          140 118 127 125 142

_After_: 140 130 135 126 134
         138 124 126 132 144

__Do the data indicate that injection of this drug changes blood pressure__

Se usará una distribución normal por indicación del problema.

In [1]:
# Se define la muestra a estudiar
before <- c(134, 122, 132, 130, 128,
            140, 118, 127, 125, 142)

after <- c(140, 130, 135, 126, 134,
           138, 124, 126, 132, 144)

# Se hace la prueba de hipótesis de las dos muestras
# (donde H_0 es que la diferencia de las  muestras es 0)
t.test(before, after)


	Welch Two Sample t-test

data:  before and after
t = -0.98112, df = 17.681, p-value = 0.3398
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -9.746757  3.546757
sample estimates:
mean of x mean of y 
    129.8     132.9 


Se concluye que no se puede inferir que hay una diferencia en la presión sanguínea.

### 57


__An ambulance service claims that at least 45 percent of its calls involve life-threatening emergencies. To check this claim, a random sample of 200 calls was selected from the service's files. If 70 of these calls involved life-threatening emergencies, is the service's claim believable at the__

__(a) 5 percent level of significance.__

__(b) 1 percent level of significance.__

Se usará una distribución normal por indicación del problema.

In [2]:
# Las proporciones dadas por el problema
p.hipotesis <- .45

# Número de valores deseados
x <- 70

#El tamaño de la muestra
n <- 200

# Se hace una prueba de hipótesis usando la distribución binomial
# significancia = 0.5
binom.test(x, n, p.hipotesis, conf.level = 0.95)


	Exact binomial test

data:  x and n
number of successes = 70, number of trials = 200, p-value = 0.004423
alternative hypothesis: true probability of success is not equal to 0.45
95 percent confidence interval:
 0.2840707 0.4204506
sample estimates:
probability of success 
                  0.35 


In [3]:
# Se hace una prueba de hipótesis usando la distribución binomial
# significancia = 0.1
binom.test(x, n, p.hipotesis, conf.level = 0.99)


	Exact binomial test

data:  x and n
number of successes = 70, number of trials = 200, p-value = 0.004423
alternative hypothesis: true probability of success is not equal to 0.45
99 percent confidence interval:
 0.2651294 0.4421971
sample estimates:
probability of success 
                  0.35 


Ni para una confianza del 95% ni del 99% se puede inferir que el 45% de las llamadas son emergencias que atenten a la vida.

## Introducción a la probabilidad y estadística
### William Mendenhall, Robert J. Beaver & Barbara M. Beaver


### 9.13 __Potencia de un antibiótico__

__Un fabricante de medicamentos dijo que la potencia media de uno de sus antibióticos fue 80%. Se probó una muestra aleatoria de n=100 cápsulas y produho una media muestral de 79.7% con una desviación estándar de 0.8%.__

__¿Los datos representan suficiente evidencia para refutar lo dicho por el fabricante?__

__(a) Exprese la hipótesis nula a ser probada.__

__(b) Exprese la hipótesis alternativa.__

__(c) Realiza una prueba estadística de la hipótesis nula y exprese su conclusión.__

La hipótesis nula es que la potencia media de los antibióticos es mayor o igual a 80%.

La hipótesis alternativa es que tienen una potencia media menor a 80%.

In [4]:
# El promedio presumido
mean <- 0.8

# El promedio experimental y su desviación
mean.experimental <- 0.797
std <- 0.008

# El tamaño de la muestra
n <- 100

In [5]:
# Se calcula el p-value
p.value <- pnorm((mean.experimental-mean)/(std/sqrt(n)))

p.value

[1] 8.841729e-05

Dado que el p.value es menor a 0.05, se rechaza la hipótesis nula y se concluye que los antibióticos ofrecen una potencia media menor a 80% con una confianza del 95%.

### 9.37 __¿Café o azul?__

__Un artículo del Washington Post expresó que casi 45% de la población de estadounidenses nace con ojos cafés, aún cuando no necesariamente siguen así. Para probar lo dicho por el periódico se seleccionó una muestra aleatoria de 80 personas y 32 de ellas tenían ojos cafés.__

__¿Hay suficiente evidencia para impugnar lo dicho por el periódico?__

In [6]:
# El promedio presumido
p.teorica <- 0.45

# Selecciones exitosas
x <-32

# El tamaño de la muestra
n <- 80

# Se realiza la prueba de hipótesis sobre la distribución binomial
binom.test(x, n, p.teorica)


	Exact binomial test

data:  x and n
number of successes = 32, number of trials = 80, p-value = 0.4317
alternative hypothesis: true probability of success is not equal to 0.45
95 percent confidence interval:
 0.2920094 0.5156231
sample estimates:
probability of success 
                   0.4 


No se cuenta con evidencia suficiente para impugnar al periódico.

### 10.5

__Las siguientes 10 observaciones son una muestra de una población normal:__

7.4 7.1 6.5 7.5 7.6

6.3 6.9 7.7 6.5 7.0

__(a) Encuentre la media y desviación estándar de estos datos.__

__(b) Encuentre un límite superior de  confianza de una cola, de 99%, para la media poblacional.__

__(c) Pruebe H0: m=7.5 contra Ha: m<7.5.__

__(d) ¿Los resultados del inciso b apoyan sus conclusiones del inciso c?__

In [7]:
# Los datos de la muestra
data <- c(7.4, 7.1, 6.5, 7.5, 7.6,
          6.3, 6.9, 7.7, 6.5, 7.0)

# Se calcula el promedio y la desviación de la muestra
(mean <- mean(data))
(std <- sd(data))
n <- length(data)

[1] 7.05

[1] 0.4994441

In [8]:
# Se calcula la cota superior
t.score <- qt(0.99, df=n-1)

(cota.superior <- mean + t.score*std/sqrt(n))

[1] 7.495613

In [9]:
# Se calcula el p-value
p.value <- pt((mean-7.5)/(std/sqrt(n)), df=n-1)

p.value

[1] 0.009556784

Como el p-value es menor a 0.01, se rechaza la hipótesis de que la media sea mayor o igual a 7.5 con una confianza del 99%.

Tanto el cálculo de la cota superior como el del p-value consiguen la misma conclusión. Esto es porque en la cota se calcula el intervalo donde hay 99% de probabilidad de encontrar el valor real y 7.5 no se encuentra en el conjunto de datos. Por su parte, la prueba de hipótesis estudia la distancia entre el promedio de la muestra y 7.5, esa distancia se mide a partir de 7.5, pero también se determina un intervalo del 99% de confianza.

En realidad, se puede afirmar que la diferencia entre los cálculos es qué valor se toma como referencia, pero el intervalo de valores en ambos casos significa lo mismo.

### 10.37

__Se realizó un experimento de diferencia pareada para comparar las medias de dos poblaciones:__

_Población 1_: 1.3 1.6 1.1 1.4 1.7
_Población 2_: 1.2 1.5 1.1 1.2 1.8

__(a) ¿Los datos dan suficiente evidencia para indicar que m1 difiere de m2?__

__(b) Encuentre el valor p aproximado para la prueba e interprete su valor.__

__(c) Encuentre un intervalo de confianza del 95% para m1-m2. Compare su interpretación del intervalo de confianza con sus resultados de la prueb en el inciso a.__

__(d) ¿Qué suposiciones se deben hacer para lograr que sus inferencias sean válidas?__

In [10]:
# Las muestras
poblacion.1 <- c(1.3, 1.6, 1.1, 1.4, 1.7)
poblacion.2 <- c(1.2, 1.5, 1.1, 1.2, 1.8)

diferencias <- poblacion.1-poblacion.2

mean.diff <- mean(diferencias)

std.diff <- sd(diferencias)

n <- length(diferencias)

In [11]:
# Se calcula el p-value con diferencias pareadas
p.value <- pt((mean.diff-0)/(std.diff/sqrt(n)), df=n-1)

p.value

[1] 0.8477206

Como el p-value (así como también su complemento) es mayor a 0.05, se infiere que no hay suficiente evidencia para afirmar que ambas muestras son diferentes. El p-value sugiere que la dispersión de los elementos de las  muestras sólo sugiere que las dos distribuciones están solapadas.

In [12]:
# Se calcula el intervalo de confianza
t.score <- qt(0.95, df=n-1)

(cota.inferior <- mean.diff - t.score*std.diff/sqrt(n))
(cota.superior <- mean.diff + t.score*std.diff/sqrt(n))

[1] -0.04870328

[1] 0.1687033

El intervalo de confianza del 95% de la distribución de la diferencia de las muestras contiene a 0, por lo que no puede afirmarse dentro del nivel de confianza que ambas muestras sean diferentes.

No parece haber suposiciones suficientes que hagan válida la prueba, visto el valor del p-value, más allá de suponer que no se trata de una distribución con forma de campana. Lo que se puede hacer es hacer muestras más grandes, en espera de disminuir el error estándar así como aumentar el tamaño de las muestras.

## Biostatistics: A Foundation for Analysis in the Health Sciences
### Wayne W. Daniel & Chad L. Cross

### 7.2.2


__A study by Thienprasiddhi et al. examined a sample of 16 subjects with open-angle glaucoma and unilateral hemifield defects. The ages (years) of the subjects were:__

62 62 68 48 51 60 51 57

57 41 62 50 53 34 62 61

__Can we conclude that the mean age of the population from which the sample may be presumed to have been drawn is less than 60 years?__

In [13]:
# Se leen los datos de la muestra
data <- c(62, 62, 68, 48, 51, 60, 51, 57,
          57, 41, 62, 50, 53, 34, 62, 61)

# Se hace una prueba t unilateral (H0: mu >= 60 años)
t.test(data, mu=60, alternative = "less")


	One Sample t-test

data:  data
t = -2.2822, df = 15, p-value = 0.01874
alternative hypothesis: true mean is less than 60
95 percent confidence interval:
     -Inf 58.82618
sample estimates:
mean of x 
  54.9375 


Como la cota superior del intervalo de confianza del 95% del promedio de la muestra no contiene a 60, se infiere que hay evidencia suficiente para inferir que la media es menor a 60 años.

### 7.2.12


__The following data are the oxygen uptakes (milimiters) during incubation of a random sample of 15 cell suspensions:__

14.0 14.1 14.5 13.2 11.2 14.0 14.1 12.2

11.1 13.7 13.2 16.0 12.8 14.4 12.9

__Do these data provide sufficient evidence at the .05 level of significance that the population mean is not 12 ml? What assumptions are necessary?__

In [14]:
# La muestra
data <- c(14.0, 14.1, 14.5, 13.2, 11.2, 14.0, 14.1, 12.2,
          11.1, 13.7, 13.2, 16.0, 12.8, 14.4, 12.9)

# Se hace la prueba t bilateral (H0: mu=12 ml)
t.test(data, mu=12)


	One Sample t-test

data:  data
t = 4.31, df = 14, p-value = 0.0007195
alternative hypothesis: true mean is not equal to 12
95 percent confidence interval:
 12.71672 14.13661
sample estimates:
mean of x 
 13.42667 


Si se supone que la muestra pertenece a una distribución aproximadamente normal (para poder usar una prueba t), entonces se infiere que, con 95% de confianza, el valor de la media es distinto de 12 ml.

### 7.2.18


__A research team is willing to assume that systolic blood pressures in a certain population of males are approximately normally distributed with a standard deviation of 16.__

__A simple random sample of 64 males from the population had a mean systolic blood pressure reading of 133.__

__At the .05 level of significance, do these data provide sufficient evidence for us to conclude that the population mean is greater than 130?.__

In [15]:
# La media, la desviación y el tamaño de la muestra
mean <- 133
std <- 16
n <- 64

# Se realiza la prueba de  hipótesis (H0: mu <= 130)
( p <- pnorm( (mean-130)/(std/sqrt(n)) ) )

[1] 0.9331928

Como el p-value del estadístico de prueba es menor a 0.95 (que correspondería a la cota superior del intervalo de confainza del 95%), se infiere que no hay evidencia suficiente para inferir que la media de la población es mayor a 130.

### 7.3.10

__Researchers wished to know if they could conclude that two populations of infants differ with respect to mean age at which they walked alone. The following data (ages in months) were collected:__

_Sample from population A_: 9.5, 10.5, 9.0, 9.75, 10.0, 13.0,
10.0, 13.5, 10.0, 9.5, 10.0, 9.75

_Sample from population B_: 12.5, 9.5, 13.5, 13.75, 12.0, 13.75,
12.5, 9.5, 12.0, 13.5, 12.0, 12.0

__What should the researchers conclude?:__

In [16]:
# Las muestras
data.A <- c(9.5, 10.5, 9.0, 9.75, 10.0, 13.0,
            10.0, 13.5, 10.0, 9.5, 10.0, 9.75)

data.B <- c(12.5, 9.5, 13.5, 13.75, 12.0, 13.75,
            12.5, 9.5, 12.0, 13.5, 12.0, 12.0)

# Se hace una prueba t bilateral con varianzas diferentes
t.test(data.A, data.B)


	Welch Two Sample t-test

data:  data.A and data.B
t = -3.1561, df = 21.969, p-value = 0.004587
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -3.0381298 -0.6285368
sample estimates:
mean of x mean of y 
 10.37500  12.20833 


Dado que el intervalo de confianza del 95% de la diferencia no contiene a 0, se infiere que la media de las muestras es diferente.

### 7.5.2

__In an article in the journal Health and Place, Hui and Bell found that among 2428 boys ages 7 to 12 years, 461 were overweight or obese.__

__On the basis of this study, can we conclude that more than 15 percent of the boys ages 7 to 12 in the sampled population are obese or overweight?__

In [17]:
# Los datos de la muestra
n <- 2428 # Tamaño
x <- 461 # Casos "favorables"

# La proporción teórica
p.teorica <- 0.15

# Se hace una prueba de proporciones
binom.test(x, n, p.teorica, alternative = "greater")


	Exact binomial test

data:  x and n
number of successes = 461, number of trials = 2428, p-value = 5.74e-08
alternative hypothesis: true probability of success is greater than 0.15
95 percent confidence interval:
 0.1768519 1.0000000
sample estimates:
probability of success 
             0.1898682 


Dado que la cota inferior del intervalo de confianza del 95% de la proporción de la muestra no contiene a 0.15, se infiere que la proporción poblacional es mayor a 0.15.

### 7.8.7

__Independent simple random samples from two strains of mice used in an experiment yielded the following measurements on plasma glucose levels following a traumatic experience:__

_Strain A_: 54; 99; 105; 46; 70; 87; 55; 58; 139; 91

_Strain B_: 93; 91; 93; 150; 80; 104; 128; 83; 88; 95; 94; 97

__Do these data provide sufficient evidence to indicate that the variance is larger in the population of strain A mice than in the population of strain B mice? Let a ¼ :05. What assumptions are necessary?__

In [18]:
# Las muestras
data.A <- c(54, 99, 105, 46, 70, 87, 55, 58, 139, 91)

data.B <- c(93, 91, 93, 150, 80, 104, 128, 83, 88, 95, 94, 97)

# Se calculan las varianzas y los tamaños de las muestras
var.A <- var(data.A)
var.B <- var(data.B)

n.A <- length(data.A)
n.B <- length(data.B)

# Se calcula el p-value de la distribución de Fisher
pf(var.A/var.B, df1 = n.A-1, df2 = n.B-1)

[1] 0.883175

In [19]:
# La proporción de las varianzas
var.A/var.B

[1] 2.141744

Como el p-value es menor a 0.95, se  infiere que no hay evidencia suficiente para afirmar que la varianza de la muestra A es mayor que la de la muestra B. Todo esto es válido si se supone que las varianzas poblacionales son, a priori, iguales.

### 7.10.2

__Given:__

H0: m <= 4.500;

HA: m > 4.500

__With__ n = 16; s = 0.020; a = 0.01

__Let__ b = 0.05 and m1 = 4.52,
__and find n and C. State the appropriate decision rule.__

In [20]:
# Los datos del ejercicio
mean.0 <- 4.500 # Hipótesis nula
std <- 0.02
n <- 16

alpha <- 0.01 # Significancia
beta <- 0.05 # El complemento de la potencia
mean.1 <- 4.520 # Hipótesis 1

# Se calculan los z scores de mu0 y mu1
z.0 <- qnorm(0.95)
z.1 <- qnorm(0.99)

# Se calcula el tamaño apropiado de la muestra
(n.apropiado <- round(( (z.0+z.1)*std/(mean.0-mean.1) )**2) )

[1] 16

El tamaño apropiado es n=16.

In [21]:
# Se calcula C
(C <- mean.0 + z.0*(std/sqrt(n.apropiado)))

[1] 4.508224

El valor crítico es C=4.508

Por lo tanto, la regla para determinar si H0 es apropiada o no es:
_Seleccione una muestra aleatoria de tamaño 16 y compute su promedio. Si el promedio es mayor a 4.508, se rechaza H0. Si el promedio es menor a 4.508, no se rechaza H0._